### PR TITLE
[18.0][MIG] product_pricelist_fixed_currency_rate: Migration to 18.0

### PR DESCRIPTION
**Pull request by BhaveshHeliconia**

### CODE REVIEW

**Comment by BhaveshHeliconia** [comment]
Hi @LoisRForgeFlow,

Could you please review this pull request when you have a moment? I would appreciate any feedback or suggestions for improvement.

Thank you for your time!

**Comment by LoisRForgeFlow** [comment]
Hi @HeliconiaSolutions 

Thanks for your contribution, before reviewing I would like that CI runs its checks, I cannot accept the workflow so we need to ask @OCA/product-maintainers to do so.

At the same time, I'm not sure what is the OCA policy regarding contributions from a company account, not sure if it is required that contributions come from individual accounts. @pedrobaeza @etobella maybe one of you can clarify this?

Kind regards,

**Comment by pedrobaeza** [comment]
There's no strict prohibition of these accounts, but first of all, the CLA should be signed, and second, I don't like these techniques, but it's just a personal POV.

**Comment by rousseldenis** [comment]
@LoisRForgeFlow Already stated about that on other PR's. Doing a PR with a global account is ok IMHO. But for commits, I tend to say they should have beeen done with an individual account.

@pedrobaeza The CLA bot was looking at commiters logins. And as only individuals are mentioned in CLAS (even there is a company one), the bot was blocking such PR's.

**Comment by pedrobaeza** [comment]
Well, apart from not seeing the CLA bot warning since a while, I think it's incorrect to not warn in these cases. A possible workaround is to put this login in a child account or similar to act like an individual.

**Comment by LoisRForgeFlow** [comment]
Yes, it is also a while since I see one of those CLA warnings

@pedrobaeza I also find it a bit weird to contribute with a company account. I see problems with responsibility and attribution; especially internally it must generate quite some headaches when the company have quite a few developers/contributors. 

> @LoisRForgeFlow Already stated about that on other PR's. Doing a PR with a global account is ok IMHO. But for commits, I tend to say they should have beeen done with an individual account.

This can be a good intermediate approach, to require that at least commits come from an individual.

**Comment by pedrobaeza** [comment]
You have GitHub organizations for pushing the branches, so there's no sense in pushing under a company account.

**Comment by LoisRForgeFlow** [comment]
Then, we are in a all-or-nothing decision and from what you said before, this has been already discussed and the decision (maybe with no full agreement) was that it is allowed to collaborate with company accounts. Am I getting it right?

**Comment by BhaveshHeliconia** [comment]
> There's no strict prohibition of these accounts, but first of all, the CLA should be signed, and second, I don't like these techniques, but it's just a personal POV.

@pedrobaeza, we’ve signed the entity CLA, which might be why you haven’t seen the CLA warning from the OCA bot.

**Comment by BhaveshHeliconia** [comment]
> Yes, it is also a while since I see one of those CLA warnings
> 
> @pedrobaeza I also find it a bit weird to contribute with a company account. I see problems with responsibility and attribution; especially internally it must generate quite some headaches when the company have quite a few developers/contributors.
> 
> > @LoisRForgeFlow Already stated about that on other PR's. Doing a PR with a global account is ok IMHO. But for commits, I tend to say they should have beeen done with an individual account.
> 
> This can be a good intermediate approach, to require that at least commits come from an individual.

@LoisRForgeFlow @pedrobaeza 
Thanks for sharing your thoughts, and we completely understand your concerns. We decided to adopt this approach because we've observed several companies doing the same, and it helps centralize contributions in a more unified manner. By using the company account, we aim to streamline the process and ensure that the entire organization shares responsibility and attribution for the work done.

That said, we're fully open to both individual and company-based contributions. We believe each approach has its merits, and we're happy to adapt based on what works best for the team and the project.

**Comment by pedrobaeza** [comment]
Using the organization GitHub remote, any in the company can push to that branch, so about the flows, you won't have any restriction.

**Comment by BhaveshHeliconia** [comment]
@pedrobaeza @LoisRForgeFlow @rousseldenis Thanks for the sharing the information.

As per our recent discussions, we have implemented the necessary changes on our side and have also signed the updated Individual CLA. Additionally, we’ve updated our profile to reflect these changes. To further support the OCA community, we’ve become proud members of OCA for 2025.

We’re excited about the opportunities for collaboration and look forward to working together with the entire OCA community to build even better open-source solutions.

**Comment by LoisRForgeFlow** [comment]
Hi @HeliconiaSolutions

Thanks for that. 

I would kindly as again @OCA/product-maintainers to accept the CI workflow of this PR, which is the first automatic review that is needed in order to move forward with this pull request.

Thanks!

File: product_pricelist_fixed_currency_rate/tests/test_product_pricelist_fixed_currency_rate.py [hunk 1]
```diff
@@ -12,85 +12,97 @@ def setUpClass(cls):
...
-            {
-                "name": "Test",
-                "currency_id": cls.env.ref("base.EUR").id,
-            }
+            [
+                {
+                    "name": "Test",
+                    "currency_id": cls.env.ref("base.EUR").id,
+                }
+            ]
```
Comment (sebalix) [inline product_pricelist_fixed_currency_rate/tests/test_product_pricelist_fixed_currency_rate.py]: IMO this change (in the whole test file) is not required (`create` accepts a dictionary), and introduces diff that could be annoying when porting commits.

**Review by sebalix — COMMENTED** [review]
LG, minor comment

**Comment by BhaveshHeliconia** [comment]
Hello @sebalix,

Thanks for the feedback! I see your point that the change might not be necessary, especially since create already accepts a dictionary. I have reverted the changes to avoid adding unnecessary diffs and potential conflicts when porting commits.

**Review by sebalix — APPROVED** [review]

**Review by LoisRForgeFlow — APPROVED** [review]
Functional and code review LGTM

**Pull request merged**